### PR TITLE
Todolist with Add,Delete  and Edit feature

### PR DESCRIPTION
**Pull request by jash-debug**
- Add a JavaScript file for the new functionality.
- Implement a function for adding a new task.
- Implement a function for deleting a task.
- Implement a function for editing task descriptions.
- Add descriptive README
- Rebuild the Main.js file
- Fix linter errors
- Make changes to delete feature

### CODE REVIEW

File: src/functionality.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    });
+
+    return item;
+  }
+
+  addTask(description) {
+    const taskObj = {
+      description,
+      completed: false,
+      index: this.tasksList.length,
```
Comment (levy002) [inline src/functionality.js:70]: - [ ] Kindly make sure the `index` starts at `1` not `0`. It means if you add the fifth task must have `index` of `5`. :wink:

File: src/functionality.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  removeTask(index) {
+    this.tasksList.splice(index, 1);
+    for (let i = index; i < this.tasksList.length; i += 1) {
+      this.tasksList[i].index = i;
+    }
+    localStorage.setItem('tasks', JSON.stringify(this.tasksList));
+    this.updateDisplay();
+  }
```
Comment (levy002) [inline src/functionality.js]: - [x] Kindly make sure when we click on the delete icon user can be able to remove the task on the list.

    chrome-capture-2022-10-12

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @jash-debug   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/functionality.js (hunk 2 above)
Comment (jash-debug) [reply src/functionality.js]: Made changes to the delete feature

**Review by sajid-munawar — APPROVED** [review]
### Hi @jash-debug , 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and description :heavy_check_mark:
 
 ## [Optional] suggestions:
 * [ ]  Please check my inline comments.

> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**